src/common.cpp

```diff
@@ -224,3 +224,27 @@ int sock_set_rate_limit(int fd, uint32_t rate_limit) {
     }
     return rc;
 }
+
+int wait_for_single_socket(int fd, int which) {
```

Comment:
Please consider moving this into `tls.cpp` (may be as static) as far as it is only place where this function is called.

Reply:
I actually put it in the common because it is general socket operation not really related to SSL. Can also be moved to tls.cpp, a matter of taste. Yet move to tls.cpp?

Reply:
Yes I understand, please move under TLS into tls.cpp until it will be used in non TLS code